controllers/discovery/initDiscoveryUserData.js

```diff
@@ -46,6 +48,7 @@ const InitDiscoveryUserData = async (req, res) => {
         ON A.user_id = B.user_id_followed AND B.user_id_follower = :userId
         WHERE A.user_id != :userId AND A.is_anonymous = false AND A.is_banned = false
         ORDER BY
+        A.karma_score DESC,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The addition of `A.karma_score DESC` in the ORDER BY clause is a good way to sort users based on their karma score. However, it's important to consider that if `karma_score` can be NULL for some users, those users will appear first when sorting in descending order. If this is not the desired behavior, you might want to handle NULL values appropriately.
```diff
-        A.karma_score DESC,
+        COALESCE(A.karma_score, 0) DESC,
```
This change will replace NULL karma scores with 0 before sorting, ensuring that users with NULL karma scores don't get unfairly prioritized.


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
mas @n-permana please follow this coderabbit recommendation